Gemfile

```diff
@@ -1,8 +1,9 @@
 source "https://rubygems.org"
 
-# Puppet 4.10.2 is the minimum version we support on Windows due to https://tickets.puppetlabs.com/browse/PUP-7383
-# On Linux we support down to 4.6
-gem "puppet", "~> #{ENV.fetch('PUPPET_VERSION', '4.10.2')}"
+# Each version of Puppet recommends a specific version of Ruby. Try to fetch the Puppet version that
+# matches our Ruby (unless PUPPET_VERSION is defined).
+matching_puppet_version = RUBY_VERSION > '2.5' ? (RUBY_VERSION > '2.7' ? '7.0.0' : '6.0.1') : '4.10.2'
```

Comment:
I would use `Gem::Version.new` here to account for weird versions (not a gem but does the trick)